### PR TITLE
[eclipse/xtext#1952] converted xtend code to java

### PR DESCRIPTION
**Pull request by cdietrich**
[eclipse/xtext#1952] converted xtend code to java
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/interpreter/AbstractConstantExpressionsInterpreter.java [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+ */
+public abstract class AbstractConstantExpressionsInterpreter {
+	@Inject
+	private ConstantOperators constantOperators;
+
+	@Inject
+	private OperatorMapping operatorMapping;
+
+	protected Object evaluate(XExpression expression, Context ctx) {
+		boolean _add = ctx.getAlreadyEvaluating().add(expression);
```
Comment (szarnekow) [inline org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/interpreter/AbstractConstantExpressionsInterpreter.java:44]: Can be inlined.

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/interpreter/AbstractConstantExpressionsInterpreter.java [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+		if (_add) {
+			Object _xtrycatchfinallyexpression = null;
```
Comment (szarnekow) [inline org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/interpreter/AbstractConstantExpressionsInterpreter.java:46]: Can be inlined

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/interpreter/AbstractConstantExpressionsInterpreter.java [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+				ctx.getAlreadyEvaluating().remove(expression);
+			}
+			return _xtrycatchfinallyexpression;
+		} else {
+			throw notConstantExpression(expression);
+		}
+	}
+
+	protected Object _internalEvaluate(XExpression expression, Context ctx) {
+		throw this.notConstantExpression(expression);
```
Comment (szarnekow) [inline org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/interpreter/AbstractConstantExpressionsInterpreter.java:59]: A couple of unnecessary `this.` qualifications in this class

**Review by szarnekow — APPROVED** [review]

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/interpreter/AbstractConstantExpressionsInterpreter.java (hunk 2 above)
Comment (cdietrich) [reply org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/interpreter/AbstractConstantExpressionsInterpreter.java:46]: forgot to readd this :(

**Pull request merged**